### DISCUSSION TITLE
[Script Editor] choose if select by ctrl+d chosses next occurence as (whole_word) or as part of a larger word

### DISCUSSION THREAD

**Discussion by Variable-ind** [godotengine/godot-proposals#7049] [Editor]
Hy :wave:, i am having a problem... there are a lot of variables/functions in the script that share some piece of word
e.g `variable` and `another_variable` both share the part (variable) in their name. so when i want to quick select (using Ctrl + d) the `variable` then a part of another_`variable` also gets selected (even though it was not highlighted by the dim white rectangle).

in short: I don't want another_`variable` to get selected :sob: .

it would be nice if i can change the behaviour of how this selection works in settings
(i-e choose to select if next Occurrence (e.g `variable`) is part of a bigger word (e.g another_`variable`) or not)

https://github.com/godotengine/godot/assets/77773850/5a38df1e-796b-4c3e-8eb2-2a3ba2055dc1

**Comment by AThousandShips** [comment]
I'd say the best would be to add a new shortcut for this behaviour, as some might rely on it, or might want both

This is handled internally in `TextEdit`, and changing it there would be very easy, just adding a flag to the arguments to `search`

**Comment by ghost** [comment]
This should use the "Whole Word" checkbox state to determine whether it applies to a whole word or a section of.

**Reply by AThousandShips** [reply]
What checkbox? This isn't for the find bar but for the shortcut, and not just the editor but any text editor, a checkbox doesn't exist as such here

**Comment by ghost** [comment]
I know it's for the shorcut, but changing the state of the checkbox in the FindBar would allow you to specify the behaviour of the shortcut.

**Reply by AThousandShips** [reply]
That would be specific and confusing imo

Also is there any specific reason you don't want to use the thread function? It really clutters the thread and you've been asked before, I just want to know

**Reply by ghost** [reply]
Why don't you go and bother someone else ?

**Reply by AThousandShips** [reply]
I am a member here at Godot, and focus on bugs and issues, and I aim to make the experience here better for everyone, your tone and attitude is unprofessional and rude, please read the [code of conduct](https://godotengine.org/code-of-conduct/) and behave accordingly, I asked politely both here and on previous instances, and you respond with immature rudeness

And I am asking because not filling up the whole screen with individual comments when continuing a thread is better makes the experience worse for everyone

**Reply by ghost** [reply]
I don't know what you mean by a thread ?  I am not trying to be rude, but you have annoyed a lot of people with your brusqueness.  I am just stating an idea, and you always jump right on top of it and shut it down.  Learn some tact & diplomacy when dealing with a community that cares about what you are doing.

**Reply by AThousandShips** [reply]
Then ask, and I explained it in one of my comments as "reply to my comments". Also how do you intend "go bother someone else" is supposed to be interpreted?

If you are upset at my behavior then either talk to me about it or the code of conduct team, but this kind of behavior is not the way to approach it.

I am sorry if my way of speaking upsets you or annoys you, but please do not speak for some "a lot of people" but for yourself. This is a text format, and there's various aspects of how people talk that come across poorly on text. What you call brusqueness is simply my way of speaking, as per the code of conduct:
> Always assume positive intent from others. Be aware that differences in culture and English proficiency make written communication more difficult than face-to-face communication and that your interpretation of messages may not be the one the author intended.

If you have an issue with the way I speak, then talk to me, or someone else, about it instead of becoming rude because I don't speak the way you want me to, I am not a mind-reader.

Me pointing out issues and questions about an idea is not "jump right on top of it and shut it down", that's what discussions about features is, if you cannot handle constructive criticism that's something you have to work with.

So, if you have a problem with the way I speak or communicate, then *talk to me*, don't just assume that I intend to be rude or brusque, you assuming intent from me when none exists, and interpreting my way of speaking as something it isn't, isn't constructive or professional, please understand that communication is two way and it's up to you to communicate your grievances, not to bottle it up and then lash out at me because I somehow don't pick up on what you are feeling.

**Reply by ghost** [reply]
I only care about making GODOT better.  To me, what you say and how you say it comes across as aggressive.  if that's not the case, then I'm sorry, I misinterpreted.  Just get on with your life & I'll get on with mine.

**Reply by AThousandShips** [reply]
And I care about that and a constructive functional environment

And one important part of that is communication, and you need to recognize that you need to communicate about these things, not just assume, that's the grease that keeps the machine going. Assuming intent and aggressiveness, when none exists, is a failure of communication on your part.

I am sorry that the way that I communicate offends you, but you also need to recognize that not everyone communicates the way you do, or with the same assumptions or expectations that you do. If you feel that you are treated rudely, or unfairly, or anything, then you need to communicate that, either to the person who you are having problems with, or the [code of conduct team](https://godotengine.org/code-of-conduct/#coc-team)

This is a fundamentally important skill in being active in a collaborative environment, and it's not helpful, or polite, to treat others based on your own assumptions.

So thank you for communicating how you feel, but please consider that there are many differences with how people communicate, and that no one is benefitted by making assumptions about the intent of others without basis.

I hope you have a good day!

**Reply by AThousandShips** [reply]
If you do not want me to interact with you I will respect that

**Reply by Zireael07** [reply]
Just sayin', I see nothing aggressive in what @AThousandShips says - both here and on other issues she communicates the general rules this repository adheres to, and does so in a way that should be clear to newcomers

**Comment by Nodragem** [comment]
This issue is a pain anytime we need to refactor to be honest, so it would be nice to agree on what to do about it 😃 !

I think the default should be whole word because after using ctrl+d only the whole word occurences are being highlighted:

<img width="760" height="376" alt="image" src="https://github.com/user-attachments/assets/75b9d261-48fd-4f23-a79b-22e3aa52933e" />

Then in terms of implementing how to swap between whole word and parts of word, there are many implementations out there already we can draw inspiration from (e.g. vs code, sublime text, zed, etc). 

What I think would be important is to sync the Find (CTRL+F)'s Whole Words with the Select Next Occurence (CTRL+D)'s Whole Words.

**Reply by kitbdev** [reply]
I don't think the default should be word mode, then it wouldn't work as expected if you only select part of a word. Though I agree that the highlights should match up.

Since other IDEs were mentioned, here is how they handle it:
VSCode select next uses the same search settings as its find bar, and even if the find bar is closed then the shortcut will temporarily show the search settings from it.
<img width="107" height="37" alt="image" src="https://github.com/user-attachments/assets/c388460d-79fe-4ccd-b522-6a90d9b99a72" />

However every other IDE I tested does not do this as far as I can tell.
Sublime Text only follows the match case option from the find bar, and only when it is open.
Instead these all have a way to skip the next occurrence - in Zed and Sublime text select next with ctrl+d and skip with ctrl+k ctrl+d, For Kate its alt+j and skip with alt+k alt+j. For Rider its alt+j and skip with F3.
In Godot its ctrl+d and skip with ctrl+alt+d.
